seatunnel-connectors-v2/connector-elasticsearch/src/main/java/org/apache/seatunnel/connectors/seatunnel/elasticsearch/source/ElasticsearchSourceReader.java

```diff
@@ -75,31 +75,29 @@ public void close() throws IOException {
 
     @Override
     public void pollNext(Collector<SeaTunnelRow> output) throws Exception {
-        synchronized (output.getCheckpointLock()) {
```

Comment:
Why remove this lock？

Reply:
Because scrolling keeps the lock from being released, when a checkpoint occurs, it will prevent the checkpoint from acquiring the lock.

Reply:
This is correct, split will prevent the checkpoint from acquiring the lock whenever the split is read, until the split read is complete.